src/communication/index.ts

```diff
@@ -1,108 +1,70 @@
-import {
-  InterfaceMessageIds,
-  InterfaceMessageEvent,
-  SentSDKMessage,
-  SDKMessageIds,
-  SDKMessageToPayload,
-  RequestId,
-  InterfaceMessageToPayload,
-  Communicator,
-} from '../types';
-import { INTERFACE_MESSAGES } from './messageIds';
+import semver from 'semver';
+import { InterfaceMessageEvent, Communicator, Methods, Response } from '../types';
+import { MessageFormatter } from './messageFormatter';
 
-class InterfaceCommunicator implements Communicator {
-  private allowedOrigins: RegExp[] = [];
+// eslint-disable-next-line
+type Callback = (response: any) => void;
+
+class PostMessageCommunicator implements Communicator {
+  private allowedOrigins: RegExp[] | null = null;
+  private callbacks = new Map<string, Callback>();
 
   constructor(allowedOrigins: RegExp[]) {
     this.allowedOrigins = allowedOrigins;
 
     window.addEventListener('message', this.onParentMessage);
   }
 
-  private isValidMessage({ origin, data }: InterfaceMessageEvent): boolean {
-    const emptyOrMalformed = !data || !data.messageId;
-    const unknownOrigin = this.allowedOrigins?.find((regExp) => regExp.test(origin)) === undefined;
+  private isValidMessage = ({ origin, data, source }: InterfaceMessageEvent): boolean => {
+    const emptyOrMalformed = !data;
+    const sentFromParentEl = source === window.parent;
     const sameOrigin = origin === window.origin;
```

Comment:
This causes issues if the interface and the safe app are hosted on the same domain. Also it should not be required after the check for the window was added.